libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -463,9 +476,9 @@ private void initLocationLayer() {
    * start / stop based on the Android lifecycle.
    */
   private void initLifecycleObservers() {
-    getLifecycle().addObserver(locationLayer);
-    getLifecycle().addObserver(locationViewModel);
-    getLifecycle().addObserver(navigationViewModel);
+    ((LifecycleOwner) getContext()).getLifecycle().addObserver(locationLayer);
```

Comment:
Same here - we are assuming the context is a `LifecycleOwner`

Reply:
I think it makes sense to throw an exception because it's gonna crash anyway if this isn't a `LifecycleOwner` and also gives us the opportunity to provide a better error message ("make sure that the provided context is a valid `LifecycleOwner`").